### PR TITLE
chore: set up initial release workflow

### PR DESCRIPTION
**Pull request by yolocs**
Part of https://github.com/abcxyz/jvs/issues/129
Missing maven release at the moment.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      - 'v*'
+env:
+  # .goreleaser.docker.yaml reads CONTAINER_REGISTRY
+  CONTAINER_REGISTRY: 'us-docker.pkg.dev/abcxyz-artifacts/docker-images'
+  WIF_PROVIDER: 'projects/1096923323432/locations/global/workloadIdentityPools/github-pool/providers/github-provider'
+  WIF_SERVICE_ACCOUNT: 'gh-access-sa@jvs-ci-test.iam.gserviceaccount.com'
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
```
Comment (sethvargo) [inline .github/workflows/release.yml:14]: I'm not sure we want to cancel in-progress releases? Seems bad to have something half-released.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          token_format: 'access_token'
+      - name: 'Authenticate to Artifact Registry'
+        uses: 'docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a' # ratchet:docker/login-action@v2
+        with:
+          username: 'oauth2accesstoken'
+          password: '${{ steps.auth.outputs.access_token }}'
+          registry: 'us-docker.pkg.dev'
+      - uses: 'goreleaser/goreleaser-action@b508e2e3ef3b19d4e4146d4f8fb3ba9db644a757' # ratchet:goreleaser/goreleaser-action@v3
+        with:
+          version: 'latest'
```
Comment (sethvargo) [inline .github/workflows/release.yml:52]: We should pin

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          args: 'release -f .goreleaser.docker.yaml --rm-dist'
+        env:
```
Comment (sethvargo) [inline .github/workflows/release.yml:54]: If the env is defined above, I don't think you need to define it again here. I think all jobs inherit it.

File: .github/workflows/release.yml (hunk 2 above)
Comment (yolocs) [reply .github/workflows/release.yml]: Manually pinned to a version

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -9,9 +9,7 @@ env:
   WIF_PROVIDER: 'projects/1096923323432/locations/global/workloadIdentityPools/github-pool/providers/github-provider'
   WIF_SERVICE_ACCOUNT: 'gh-access-sa@jvs-ci-test.iam.gserviceaccount.com'
 
-concurrency:
-  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
-  cancel-in-progress: true
+concurrency: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
```
Comment (sethvargo) [inline .github/workflows/release.yml:12]: Also add a comment about why we don't cancel-in-progress please.

**Review by sethvargo — APPROVED** [review]

**Pull request merged**